cocotb/regression.py

```diff
@@ -406,9 +400,16 @@ def _result_was():
 
         return result_pass, sim_failed
 
-    def execute(self):
-        self._running_test = cocotb.regression_manager.next_test()
-        if self._running_test:
+    def execute(self) -> None:
+        while True:
+            self._test = self.next_test()
+            if not self._test:
```

Comment:
```suggestion
            if self._test is None:
```
is perhaps a little clearer